Samples for Extending Pentaho/Reference Implementations/Security/SAML 2.0/pentaho-saml/src/main/java/org/pentaho/platform/spring/security/saml/PentahoSamlUserRoleListService.java

```diff
@@ -1,38 +1,49 @@
 package org.pentaho.platform.spring.security.saml;
 
-
-import org.pentaho.platform.api.engine.IUserRoleListService;
-import org.pentaho.platform.api.mt.ITenant;
-import org.pentaho.platform.engine.core.system.PentahoSystem;
-import org.springframework.util.Assert;
-
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 
+import org.pentaho.platform.api.engine.IUserRoleListService;
+import org.pentaho.platform.api.mt.ITenant;
+import org.pentaho.platform.engine.core.system.PentahoSystem;
+import org.pentaho.platform.spring.security.saml.groups.PentahoSamlNativeUserRoleListService;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import org.springframework.util.Assert;
 
-public class PentahoSamlUserRoleDetailsService implements IUserRoleListService {
+public class PentahoSamlUserRoleListService implements IUserRoleListService {
 
-  private static Logger logger = LoggerFactory.getLogger( PentahoSamlUserRoleDetailsService.class );
+  private static Logger logger = LoggerFactory.getLogger( PentahoSamlUserRoleListService.class );
 
   private static final String PROVIDER_NAME = "providerName";
 
+  //Delegate user role list service
   private IUserRoleListService service;
-  String selectedAuthorizationProvider;
+  private String selectedAuthorizationProvider;
+
+  private PentahoSamlNativeUserRoleListService samlUserRoleListService;
+  private String samlId;
 
-  public PentahoSamlUserRoleDetailsService( IUserRoleListService service ) {
+  /*
+  public PentahoSamlUserRoleListService( IUserRoleListService service ) {
     this.service = service;
   }
+  */
 
-  public PentahoSamlUserRoleDetailsService( String selectedAuthorizationProvider ) {
+  public PentahoSamlUserRoleListService( String selectedAuthorizationProvider ) {
 
     setSelectedAuthorizationProvider( selectedAuthorizationProvider );
     Assert.notNull( selectedAuthorizationProvider );
   }
```

Comment:
while we're on it, let's change the order of these 2 lines ( i.e. have the assertion made prior to the setter )